### PR TITLE
CLOUD-625 - Create chaos-mesh e2e-tests

### PR DESCRIPTION
**Pull request by tplavcic**
[CLOUD-625](https://jira.percona.com/browse/CLOUD-625) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| arbiter | passed |
| operator-self-healing | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| operator-self-healing-chaos | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/8fc59d916e0658fd32b40329c68c73c9ce680fc7
image: `perconalab/percona-server-mongodb-operator:PR-669-8fc59d91`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| operator-self-healing | passed |
| upgrade | passed |
| liveness | passed |
| operator-self-healing-chaos | passed |
| upgrade-sharded | passed |
| smart-update | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/2ae9319917ef42d67e3a96f9aaa46eaccd77f11f
image: `perconalab/percona-server-mongodb-operator:PR-669-2ae93199`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| operator-self-healing-chaos | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/53c95f471cc6a41c8f20a0cbbb28f65ee3286448
image: `perconalab/percona-server-mongodb-operator:PR-669-53c95f47`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| one-pod | passed |
| demand-backup | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | failed |
| operator-self-healing | passed |
| demand-backup-sharded | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing-chaos | failed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/6a9a83de3b3eec58ed28882c5c87ad37ee15fcf1
image: `perconalab/percona-server-mongodb-operator:PR-669-6a9a83de`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| storage | passed |
| smart-update | passed |
| self-healing | passed |
| version-service | passed |
| self-healing-chaos | failed |
| users | passed |
| data-sharded | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/6a9a83de3b3eec58ed28882c5c87ad37ee15fcf1
image: `perconalab/percona-server-mongodb-operator:PR-669-6a9a83de`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| operator-self-healing-chaos | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/95c1027194d165fe9339f67b2e0a81971d9b1281
image: `perconalab/percona-server-mongodb-operator:PR-669-95c10271`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| operator-self-healing-chaos | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/afb39a3ad7dbf63555de32f67318fb7bc05d6d12
image: `perconalab/percona-server-mongodb-operator:PR-669-afb39a3a`

**Review by delgod — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -305,7 +305,9 @@ pipeline {
                         CreateCluster('selfhealing')
                         runTest('storage', 'selfhealing')
                         runTest('self-healing', 'selfhealing')
+                        runTest('self-healing-chaos', 'selfhealing')
                         runTest('operator-self-healing', 'selfhealing')
```
Comment (cap1984) [inline Jenkinsfile:309]: no self-healing-advanced-chaos ?

Comment (tplavcic) [reply Jenkinsfile:309]: it doesn't seem to exist for PSMDB, only for PXC, chaos tests at the moment were created as pairs for pumba tests

File: e2e-tests/functions [hunk 2]
```diff
@@ -335,6 +344,26 @@ deploy_minio() {
...
 }
 
+deploy_chaos_mesh() {
+	desc 'install chaos-mesh'
+	helm del chaos-mesh --namespace chaos-testing || :
+	kubectl_bin delete ns chaos-testing || :
+	helm repo add chaos-mesh https://charts.chaos-mesh.org
+	kubectl_bin create ns chaos-testing
+	if version_gt "1.20"; then
+		helm install chaos-mesh chaos-mesh/chaos-mesh --namespace=chaos-testing --set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock --set dashboard.create=false
```
Comment (cap1984) [inline e2e-tests/functions:354]: I believe we need to use fixed version here.

File: e2e-tests/operator-self-healing-chaos/run [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+		| yq w - "metadata.namespace" "chaos-testing" \
+		| yq d - "spec.selector.pods.test-namespace" \
+		| yq w - "spec.selector.pods.$namespace[0]" "$init_pod" \
+		| kubectl_bin apply -f-
+	sleep 10
+
+	desc 'check if operator works fine: scale down from 5 to 3'
+	kubectl_bin get psmdb -o yaml ${cluster%%-rs0} \
+		| sed -e 's/size: 5/size: 3/' \
+		| kubectl_bin apply -f-
```
[the remark this answers is not in the document]
Comment (tplavcic) [reply e2e-tests/operator-self-healing-chaos/run:35]: what's the typo?

File: e2e-tests/functions (hunk 2 above)
Comment (tplavcic) [reply e2e-tests/functions]: Thanks! fixed the chaos-mesh chart version to `0.5.1`

File: e2e-tests/operator-self-healing-chaos/run (hunk 3 above)
[the remark this answers is not in the document]
Comment (cap1984) [reply e2e-tests/operator-self-healing-chaos/run:35]: `-f-`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| demand-backup-sharded | passed |
| arbiter | passed |
| rs-shard-migration | passed |
| service-per-pod | failed |
| operator-self-healing | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| operator-self-healing-chaos | passed |
| pitr | passed |
| liveness | failed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/7e6009306ff25c15d5c88ec94e85deda92e317b9
image: `perconalab/percona-server-mongodb-operator:PR-669-7e600930`

**Review by delgod — APPROVED** [review]

**Review by hors — CHANGES_REQUESTED** [review]
@tplavcic  could you please re-check self-healing-chaos test. It does not work for me: 
The error is:
```
> cat /var/folders/kw/j3p4xcxj0_g1z7nnm9f8tqzm0000gn/T/tmp.pJDXtlaQ
Error from server (***@percona.com is forbidden on namespace self-healing-chaos-22773): error when creating "STDIN": admission webhook "vauth.kb.io" denied the request: ***@percona.com is forbidden on namespace self-healing-chaos-22773
```

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| operator-self-healing | passed |
| upgrade | passed |
| liveness | passed |
| upgrade-sharded | passed |
| operator-self-healing-chaos | passed |
| smart-update | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/95c6b27a8a50c458ad277c9b795fde691b78ef70
image: `perconalab/percona-server-mongodb-operator:PR-669-95c6b27a`

File: e2e-tests/functions [hunk 4]
```diff
@@ -335,6 +344,26 @@ deploy_minio() {
         /usr/bin/aws --endpoint-url http://minio-service:9000 s3 mb s3://operator-testing'
 }
 
+deploy_chaos_mesh() {
+	desc 'install chaos-mesh'
+	helm del chaos-mesh --namespace chaos-testing || :
+	kubectl_bin delete ns chaos-testing || :
+	helm repo add chaos-mesh https://charts.chaos-mesh.org
+	kubectl_bin create ns chaos-testing
+	if version_gt "1.20"; then
```
Comment (hors) [inline e2e-tests/functions:353]: in GKE containerd  is used started from 1.19. please fix it in all files.

Comment (tplavcic) [reply e2e-tests/functions]: Thanks! Fixed!

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| operator-self-healing-chaos | passed |
| upgrade-sharded | passed |
| users | passed |
| pitr | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/705396bd922591e33df17ca0e5ca8902d9d0e382
image: `perconalab/percona-server-mongodb-operator:PR-669-705396bd`

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | failed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| operator-self-healing | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| liveness | passed |
| operator-self-healing-chaos | passed |
| smart-update | passed |
| pitr | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/e9eaf4a414b1c80f0fc85401e053c5acb61ee4c6
image: `perconalab/percona-server-mongodb-operator:PR-669-e9eaf4a4`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| one-pod | passed |
| storage | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| arbiter | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/e9eaf4a414b1c80f0fc85401e053c5acb61ee4c6
image: `perconalab/percona-server-mongodb-operator:PR-669-e9eaf4a4`

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| operator-self-healing | passed |
| upgrade | passed |
| liveness | passed |
| upgrade-sharded | passed |
| smart-update | passed |
| operator-self-healing-chaos | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/669/commits/e7720a27ec0a5eaf293d90354c23e9da212ef958
image: `perconalab/percona-server-mongodb-operator:PR-669-e7720a27`

**Pull request merged**